packages/pwa-kit-react-sdk/src/ssr/universal/components/_app-config/index.jsx

```diff
@@ -51,6 +51,26 @@ class AppConfig extends React.Component {
         return undefined
     }
 
+    // TODO: Fix vagueness of this description.
+    /**
+     * Freezes the request into the rendered HTML as `__ORIGINAL_REQUEST__`.
+     * It's important to note that this function is only called once on the server-side.
+     *
+     * @param req - the request object.
+     * @param res - the response object.
+     * @return {Object} - the application state as an object, which will be
+     *    serialized into JSON and embedded in the page HTML.
+     */
+    // eslint-disable-next-line no-unused-vars
+    static freezeRequest(req, res) {
```

Comment:
Chatted with Ben IRL but leave a record here for discussion:

I'm wondering do we want to be a bit opinionated and just freeze every serializable key from the `req` object? It feels a bit that this API doesn't provide that much more flexibility but scarifies deepening the learning curve of the SDK.

Personally I think just have the serialized `req` object available is more than enough and developers don't have to learn an extra API method.